arch/arc/core/fast_irq.S

```diff
@@ -81,12 +78,12 @@ SECTION_FUNC(TEXT, _firq_enter)
 	lr r25, [_ARC_V2_LP_END]
 #endif
 
-	ld r1, [exc_nest_count]
-	add r0, r1, 1
-	st r0, [exc_nest_count]
-	cmp r1, 0
-
-	bgt.d  firq_nest
+	lr r0, [_ARC_V2_AUX_IRQ_ACT]
+	and r0, r0, 0xffff
+	ffs r1, r0
+	fls r2, r0
+	cmp r1, r2
+	bne.d firq_nest
```

Comment:
That sequence of instructions seems to be repeated multiple times:
```
	lr r0, [_ARC_V2_AUX_IRQ_ACT]
	and r0, r0, 0xffff
	ffs r1, r0
	fls r2, r0
	cmp r1, r2
```

So maybe have it as a macro instead of copy-pasting many times?

Reply:
good idea, already replaced with a macro